### ISSUE TITLE
Ability to synchronize the song to the grid

### ISSUE THREAD

**Issue by SCSR-is-too-short-username** [Caeden117/ChroMapper#498]
**What is your request?**
Currently the song time offset seems to be disabled, however there is no alternate way of synchronizing the song to the grid. This has a significant negative impact on the user experience. Other editors I've used still has the offset and the maps work fine in game, so I have no idea why it is disabled in the first place...

**Comment by Top-Cat** [comment]
> there is no alternate way of synchronizing the song to the grid
Yes there is. Follow the bsmg guides: https://bsmg.wiki/mapping/#mapping-quick-start
> the maps work fine in game
Except they don't. Just because you haven't noticed the issues doesn't mean they don't exist.

**Comment by SCSR-is-too-short-username** [comment]
> Except they don't. Just because you haven't noticed the issues doesn't mean they don't exist.

Well, what issue does it have?
Anyway, having some form of syncing built in the editor would still be nice, even if you can't use the song time offset. Having to do it in another program is a bit of a pain.

**Comment by LightAi39** [comment]
Note syncing breaks at any other speed than 100%. Beat Games have said they will not fix it, and have thus also disabled it in the official editor. It's a deprecated feature and would break any maps timings when using SS, FS, or SF.

**Comment by Caeden117** [comment]
There has been small discussions about this in the past. From my understanding, syncing the song to the start beat would be sometime after [BPM detection](https://github.com/Caeden117/ChroMapper/tree/bpm-detection), which is currently blocked by the [Unity 2021.3 upgrade](https://github.com/Caeden117/ChroMapper/tree/2021.3)
